### PR TITLE
1123 superset persistent

### PR DESCRIPTION
**Pull request by kmagida110**
## Background
This PR updates CircleCI to launch an instance of Superset on the designated instance with an attached, persistent Postgres instance. Instance can be found here: http://ec2-3-134-85-99.us-east-2.compute.amazonaws.com/

## GitHub Issue
https://github.com/ctoec/data-collection/issues/1123

## Validation Plan
Visual validation of running Superset and confirm ability to create charts, dashboards and upload data.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -42,23 +45,49 @@ jobs:
...
       - run:
-          name: 'Start SuperSet (with examples)'
+          name: 'Start SuperSet'
           command: |
-            ssh ec2-user@ec2-3-134-85-99.us-east-2.compute.amazonaws.com "cd superset && SUPERSET_LOAD_EXAMPLES=yes docker-compose up -d"
+            ssh ec2-user@ec2-3-134-85-99.us-east-2.compute.amazonaws.com "cd docker && docker-compose down && docker-compose up -d"
+      - run:
+          name: 'Copy Favicon'
+          command: |
+            ssh ec2-user@ec2-3-134-85-99.us-east-2.compute.amazonaws.com "docker cp docker/images/favicon.ico superset:/app/superset/static/assets/images/favicon.png"
```
Comment (jlhogan) [inline .circleci/config.yml:69]: can't this just be accomplished with a mount in the docker compose? Why does it need to be manually copied into running docker ? EDIT: why is this required looks like it is indeed mounted in docker-compose project

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+1. Upgrade and initialize database from command line of Superset instance
+   - `docker exec -it superset superset db upgrade && docker exec -it superset superset init`
+1. Create admin user from command line of Superset instance
+   - `docker exec -it superset superset fab create-admin \
+                 --username USERNAME \
+                --firstname FIRSTNAME \
+                --email EMAIL \
+                --password ********`
+1. Add Database to Superset UI
```
Comment (jlhogan) [inline docker/README.md:30]: what does this step mean?

File: docker/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+   - Add database through database UI with connection string `SQLALCHEMY_DATABASE_URI='postgresql+psycopg2://superset_admin:$SUPERSET_DB_PASS@superset_db/superset`
+   - Enable CSV upload and all SQL Lab settings
```
Comment (jlhogan) [inline docker/README.md:32]: this is done thru UI? weird that it's not configurable.

File: .circleci/config.yml [hunk 4]
```diff
@@ -42,23 +45,49 @@ jobs:
       - checkout
       - add_ssh_keys
       - run:
-          name: 'Remove previous SuperSet GitHub clone'
+          name: 'Add DB Password to config files'
           command: |
-            ssh ec2-user@ec2-3-134-85-99.us-east-2.compute.amazonaws.com "sudo rm -rf superset"
+            SUPERSET_DB_PASS=$(aws secretsmanager get-secret-value --secret-id /data-viz/db/db_pass --query 'SecretString' --output text) && echo -e "SQLALCHEMY_DATABASE_URI='postgresql+psycopg2://superset_admin:$SUPERSET_DB_PASS@superset_db/superset'" >> docker/config/superset_config.py && echo -e "POSTGRES_PASSWORD=$SUPERSET_DB_PASS" >> docker/.env
```
Comment (jlhogan) [inline .circleci/config.yml:50]: why can't the env var just be set for docker, and then config defines `SQLALCHEMY_DATABASE_URI` with env var interpolation?

File: docker/README.md (hunk 3 above)
Comment (kmagida110) [reply docker/README.md:32]: This is the database for the actual data, the superset metadata is the one that is configurable. We have them as the same DB for now. Their theory is that the data sources are all UI based.

Comment (jlhogan) [reply docker/README.md:32]: sorry i mean the step of `enable CSV upload + all SQL Lab settings`  -- is this a manual step done thru the UI?

File: .circleci/config.yml (hunk 1 above)
Comment (kmagida110) [reply .circleci/config.yml:69]: I think the copy needs to happen after the app starts up since it is populated in the startup. I tried not having this and the old logo came up. I think it needs to be a running container.

File: docker/README.md (hunk 2 above)
Comment (kmagida110) [reply docker/README.md:30]: Is this better? `Add Database to store OEC data through Superset UI`

File: .circleci/config.yml (hunk 1 above)
Comment (jlhogan) [reply .circleci/config.yml:69]: >since it is populated in the startup 
not sure what that means. If it doesn't work without it, then that's that i guess but doesnt really make sense to me!

File: docker/README.md (hunk 3 above)
Comment (kmagida110) [reply docker/README.md:32]: Those options are on the datasource level so they push you to the UI as well. Some DBs may not allow upload or SQL Lab.

File: .circleci/config.yml (hunk 4 above)
Comment (kmagida110) [reply .circleci/config.yml]: Updated

File: docker/README.md (hunk 3 above)
Comment (jlhogan) [reply docker/README.md:32]: gotcha!

File: .gitignore [hunk 5]
```diff
@@ -130,3 +130,6 @@ dmypy.json
 
 # Data folders
 /src/data_integration/july_2020/data/
+
+*.DS_Store
+!docker/.env
```
Comment (JordanGuinn) [inline .gitignore:135]: I think if we removed everything in this `.gitignore` except these last 4 lines, there wouldn't be a need for this last line.  I feel like most of our `.gitignore` doesn't actually pertain to anything we're building.

File: docker/README.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+# Superset creation from Dockerfile
```
Comment (JordanGuinn) [inline docker/README.md:1]: Is there any reason this can't live in the main README?  Do we need two READMEs?

File: .circleci/config.yml (hunk 1 above)
Comment (JordanGuinn) [reply .circleci/config.yml:69]: This is only necessary for the favicon, not the logo?  It's fine for now, but when we inevitably pull more static assets into our application, we'll need to figure out what's actually going on here

**Review by JordanGuinn — APPROVED** [review]
Looks fine overall, save for the few questions I left in a couple places

**Comment by JordanGuinn** [comment]
Also, if it hasn't happened already, validation probably ought include multiple successive CircleCI runs without issue.

File: .gitignore (hunk 5 above)
Comment (kmagida110) [reply .gitignore:135]: OK, I'll pull it out, that was just the standard Python gitignore and I'll just use what I need from it.

File: docker/README.md (hunk 6 above)
Comment (kmagida110) [reply docker/README.md:1]: Nope!

**Comment by kmagida110** [comment]
@JordanGuinn I've validation many times, mostly to try to get the logo as close to the right size as possible.

**Review by jlhogan — APPROVED** [review]
